bibernate-core/src/main/java/com/bobocode/petros/bibernate/session/CacheableSession.java

```diff
@@ -60,11 +62,15 @@ public <T> Query<T> createQuery(String sql, Class<T> type) {
 
     @Override
     public Transaction getTransaction() {
-        return super.getTransaction();
+        return jdbcQueryManager.getTransaction();
     }
 
     @Override
     public void close() {
-        super.close();
+        if (!getTransaction().isClosed()) {
+            getTransaction().commit();
+        }
+        //TODO: think if there's another way to close connection. Add closeConnection to Transaction?
+        getTransaction().rollback();
```

Comment:
This code should be executed in case when commit failed. Here it looks like if block is executed (commit transaction) it will rollback with the next step. Is it the desired behavior? Could the code that is responsible for connection closing be moved to DefaultSession.close() ?